core/src/trezor/pin.py

```diff
@@ -35,9 +49,12 @@ def show_pin_timeout(seconds: int, progress: int, message: str) -> bool:
     else:
         remaining = _previous_remaining
 
-    if progress == 0 or _progress_layout is None:
+    # create the layout if it doesn't exist yet or should be started again
+    if _progress_layout is None or (progress == 0 and not _started_with_empty_loader):
         _progress_layout = pin_progress(message, description=remaining or "")
+
```

Comment:
my guess is adding `_started_with_empty_loader = False` here will get rid of the rendering artifact

Reply:
Right, adding that in [11c5e09](https://github.com/trezor/trezor-firmware/pull/2688/commits/11c5e0950446c20481b58853cd8b9c56150afcde) helped

Reply:
it should be outside the `if` though. it actually surprised me that this worked; apparently it is because the `if seconds == 0` clear the `_progress_layout`, but technically speaking that condition _may_ not happen at all, in which case you'd get the artifact the next time someone ""restarts"" after not finishing the previous loader.

Reply:
Moved outside of `if` and included a comment in a force-push together with rebase on `master`